src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -37,3 +47,19 @@ def validate_reservations_summaries(namespace):
         raise CLIError("usage error: --grain  can be either daily or monthly.")
     if data_grain == 'daily' and (not namespace.start_date or not namespace.end_date):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied for daily grain.")
+
+
+def validate_budget_parameters(namespace):
+    """lowercase the time grain for comparison"""
+
+    budgetcategory = namespace.category.lower()
+    if budgetcategory != 'cost' and budgetcategory != 'usage':
+        raise CLIError("usage error: --parameters.category must be set to either cost or usage")
```

Comment:
There's no way to have an option names `--parameters.category`.  Why is this not using the CLI's built-in enum validation?

Reply:
This category is a field of budget model. We put it here to remind users of the field